frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -48,40 +144,211 @@ class Chart extends Component<Props, State> {
+                    return;
+                }
+                const { utcYear, from, to } = this.getUTCRange();
+                from.setUTCFullYear(utcYear - 1);
+                this.chart && this.chart.timeScale().setVisibleRange({
+                    from: from.getTime() / 1000 as UTCTimestamp,
+                    to: to.getTime() / 1000 as UTCTimestamp,
+                });
+            }
+        );
+    }
+
+    private displayAll = () => {
+        if (this.state.source !== 'daily' && this.lineSeries && this.props.dataDaily) {
+            this.lineSeries.setData(this.props.dataDaily);
+        }
+        this.setState(
+            { display: 'all', source: 'daily' },
+            () => {
+                if (!this.chart) {
+                    return;
+                }
+                this.chart.timeScale().fitContent();
+            }
+        );
+    }
+
+    private calculateChange = () => {
+        const data = this.props[this.state.source === 'daily' ? 'dataDaily' : 'dataHourly'];
+        if (!data || !this.chart || !this.lineSeries) {
+            return;
+        }
+        const { barsBefore } = this.lineSeries.barsInLogicalRange(this.chart.timeScale().getVisibleLogicalRange() as any) as any;
+        const rangeFrom = Math.max(Math.floor(barsBefore), 0);
+        if (!data[rangeFrom]){
+            // when data series have changed it triggers subscribeVisibleLogicalRangeChange
+            // but at this point the setVisibleRange has not executed what the new range
+            // should be and therefore barsBefore might still point to the old range
+            // so we have to ignore this call and expect setVisibleRange with correct range
+            this.setState({ difference: 0, diffSince: '' });
+            return;
+        }
+        const valueFrom = data[rangeFrom].value;
+        const valueTo = this.state.currentTotal;
+        const valueDiff = valueTo ? valueTo - valueFrom : 0;
+        this.setState({
+            difference: ((valueDiff / valueFrom) * 100),
+            diffSince: `${data[rangeFrom].value.toFixed(2)} / ${new Date(Number(data[rangeFrom].time) * 1000)}`
+        });
     }
 
     public render(
-        { t, data }: RenderableProps<Props>,
-        { }: State,
+        { t, dataDaily, fiatUnit }: RenderableProps<Props>,
+        { currentTotal, difference, diffSince, display }: State,
     ) {
+        if (!dataDaily) {
+            return (
+                <p>
+                    {t('chart.dataMissing')}
+                </p>
+            );
+        }
+
+        const diff = difference && Number.isFinite(difference) ? (
+            <span className={styles[difference < 0 ? 'down' : 'up']} title={diffSince}>
+                <span className={styles.arrow}>
+                    {difference < 0 ? (<ArrowUp />) : (<ArrowDown />)}
+                </span>
+                <span className={styles.diffValue}>
+                    {formatAsCurrency(difference)}
```

Comment:
rename `formatAsCurrency` to `formatNumber()` or sth? `difference` is not a currency value.

Reply:
I renamed the function at the import. Will be replaced when support different formatting

Reply:
Why not rename it at the source? We could change how the function works and miss that it also formats non-currency numbers.